go-controller/pkg/clustermanager/egressip_controller.go

```diff
@@ -1047,6 +1047,64 @@ func (eIPC *egressIPClusterController) reconcileEgressIP(old, new *egressipv1.Eg
 	return nil
 }
 
+// syncEgressIPs This method takes care syncing stale data in the egress ip status
+// with cloud private ip config upon master reboot cases. cloud private ip config
+// entry would have been deleted when master was down whereas egress ip status
+// was not updated for the deleted entry in an error scenario. Hence this method
+// ensures egress ip status is upto date with available cloud private ip config
+// entry.
+func (eIPC *egressIPClusterController) syncEgressIPs(ips []interface{}) error {
```

Comment:
Wouldn't it make sense for this to be a sync function of the CloudPrivateIPConfigs instead of the EgressIPs? As an added benefit we would have everything we need in the informer cache.